proof/capDL-api/Retype_DP.thy

```diff
@@ -537,14 +533,9 @@ lemma mark_tcb_intent_error_has_children[wp]:
   by (wpsimp simp: has_children_def is_cdt_parent_def mark_tcb_intent_error_def update_thread_def
                    set_object_def)
 
-crunch cdt[wp]: corrupt_frame "\<lambda>s. P (cdl_cdt s)"
-(wp:select_wp simp:crunch_simps corrupt_intents_def)
-
-crunch cdt[wp]: corrupt_tcb_intent "\<lambda>s. P (cdl_cdt s)"
-(wp:select_wp simp:crunch_simps corrupt_intents_def)
-
-crunch cdt[wp]: corrupt_ipc_buffer "\<lambda>s. P (cdl_cdt s)"
-(wp:select_wp simp:crunch_simps corrupt_intents_def)
+crunches corrupt_frame, corrupt_tcb_intent, corrupt_ipc_buffer
+  for cdt[wp]: "\<lambda>s. P (cdl_cdt s)"
```

Comment:
Off-topic, but would there be any benefit to making crunches look for a free `s` in the term and adding a `λs.` in front? Would that be something that makes things nicer, or more confusing?

Reply:
Hmm, interesting. It would definitely save a lot of repetition and make things a bit nicer for experienced users but my first reaction is that it would be more confusing for people who are encountering it for the first time.

Reply:
I don't think it would work, unless you encode the name `s` as something special. If you look at the statements, `P` is a legitimate free variable in the term and would look the same as an `s` to Isabelle unless you start analysing for types etc.